### PR TITLE
Deal with changes to the WebRender API

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because they should not change behavior.

The WebRender display list is now similar to the Servo display list,
which simplifies the conversion.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14200)

### CODE REVIEW

File: components/servo/Cargo.toml [hunk 1]
```diff
@@ -100,3 +100,7 @@ android_glue = "0.2"
 
 [target.'cfg(not(target_os = "windows"))'.dependencies]
 gaol = {git = "https://github.com/servo/gaol"}
+
+[replace]
+"webrender:0.8.0" = { path = '/home/martin/work/mozilla/webrender/webrender/' }
```
Comment (Ms2ger) [inline components/servo/Cargo.toml]: Uh...

**Review by Ms2ger — DISMISSED** [review]

File: components/servo/Cargo.toml (hunk 1 above)
Comment (mrobinson) [inline components/servo/Cargo.toml]: Okay. I've cleaned this up. Sorry, this wasn't ready for review yet.

**Comment by mrobinson** [comment]
Okay, this should be ready to review now. r? @glennw, @pcwalton, or anyone else who feels confident.

**Comment by nox** [comment]
@mrobinson There was no shader change in that bump?

**Comment by mrobinson** [comment]
@nox No, no shaders were updated as far as I can tell.

**Comment by glennw** [comment]
@bors-servo p=10

(Until this lands it's complicated working with a local WR override due to the version / API mismatch).

**Comment by glennw** [comment]
@bors-servo retry

 * https://github.com/servo/servo/issues/12797

**Comment by glennw** [comment]
The opacity one is a known intermittent. This passed a try run previously, so I'm not sure out the other failure. Building locally to test now.

**Comment by glennw** [comment]
@bors-servo retry
* #13360

**Comment by glennw** [comment]
@mrobinson The iframe/overflow.html test seems permanent - not sure what happened between the successful try run and this failure. Perhaps something merged in between that broke things?

**Comment by mrobinson** [comment]
@glennw I've run tests again locally after rebasing and I'm not seeing this failure. ☹

**Comment by mrobinson** [comment]
I'm not really sure how my changes could have affected this test, but I'm going to keep investigating. It could be that this is just a timing issue and the test was already brittle.

**Comment by mrobinson** [comment]
Going to try tests again, just to verify that this isn't an issue that is caused by r+ versus try.

**Comment by mrobinson** [comment]
One failure and it looks like it is https://github.com/servo/servo/issues/13776. Perhaps we can try to land this again?

**Comment by glennw** [comment]
@bors-servo r+ retry
* #13776

**Comment by mrobinson** [comment]
@bors-servo retry
   * #13360

**Comment by mrobinson** [comment]
I extended the reftest-wait workaround to the other failing iframe test, perhaps it would be okay to try landing this one more time?

**Comment by emilio** [comment]
@bors-servo r=emilio,glennw

 * https://github.com/servo/servo/issues/14247

**Comment by emilio** [comment]
I think the reftest-wait will help nothing, but it doesn't hurt either

**Pull request merged**